src/rate_limiter.cc

```diff
@@ -91,6 +92,44 @@ RateLimiter::RegisterModelInstance(
   return Status::Success;
 }
 
+Status
+RateLimiter::UnregisterModelInstance(TritonModelInstance* triton_model_instance)
+{
+  std::lock_guard<std::mutex> lk1(model_ctx_mtx_);
+  std::lock_guard<std::mutex> lk2(model_instance_ctx_mtx_);
+
+  const TritonModel* model = triton_model_instance->Model();
+
+  auto& model_context = model_contexts_[model];
+  auto& model_instances = model_instance_ctxs_[model];
+  auto i_it = model_instances.find(triton_model_instance);
+  if (i_it != model_instances.end()) {
+    i_it->second->WaitForRemoval();
+    if (!ignore_resources_and_priority_) {
+      resource_manager_->RemoveModelInstance(i_it->second.get());
+    }
+    model_context.RemoveInstance(i_it->second.get());
+    model_instances.erase(i_it);
+  }
+
+  if (!ignore_resources_and_priority_) {
+    RETURN_IF_ERROR(resource_manager_->UpdateResourceLimits());
```

Comment:
We can skip these update if there is only a model instance count change.  UpdateResourceLimits will not generate new results in such cases.
Add DLIS-4821 as the proposed optimization as a TO-DO.

Reply:
Updated: [Add links to JIRA](https://github.com/triton-inference-server/core/pull/189/commits/9f65d1267f872ff863c7b0d66910a159e6a3f561)